gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageGrpcReadChannel.java

```diff
@@ -387,6 +403,18 @@ public int read(ByteBuffer byteBuffer) throws IOException {
     return bytesRead;
   }
 
+  private boolean isByteBufferBeyondCurrentRequestRange(ByteBuffer byteBuffer) {
+    long effectivePosition = position + bytesToSkipBeforeReading;
+    // current request does not have a range or this is the first request
+    if (contentChannelEndOffset == -1) {
+      return false;
+    }
+    if ((effectivePosition + byteBuffer.remaining()) > (contentChannelEndOffset)) {
```

Comment:
can be simplified: `return (effectivePosition + byteBuffer.remaining()) > (contentChannelEndOffset);`